vcd/resource_vcd_vapp_vm.go

```diff
@@ -96,13 +107,70 @@ func resourceVcdVAppVm() *schema.Resource {
 				Default:  true,
 			},
 			"network_href": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: "In favor of networks",
+			},
+			"networks": {
+				ConflictsWith: []string{"ip", "network_name"},
+				ForceNew:      true,
+				Optional:      true,
+				Type:          schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"orgnetwork": {
+							ForceNew: true,
+							Required: true,
+							Type:     schema.TypeString,
+						},
+						"ip": {
+							Computed: true,
+							ForceNew: true,
+							Optional: true,
+							Type:     schema.TypeString,
+							ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+								if net.ParseIP(val.(string)) == nil {
+									errs = append(errs, fmt.Errorf("IP must be valid, got: %d", val.(string)))
```

Comment:
```suggestion
									errs = append(errs, fmt.Errorf("IP must be valid, got: %s", val.(string)))
```

Reply:
Fixed as in replaced it with SingleIP() function